### PR TITLE
improve error reporting for parallel downloading of metadata

### PR DESCRIPTION
**Pull request by mhatina**

### CODE REVIEW

File: librepo/metadata_downloader.c [hunk 1]
```diff
@@ -18,7 +18,9 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
-#define _GNU_SOURCE // for SA_RESTART
+#define _POSIX_SOURCE
+#define _DEFAULT_SOURCE
```
Comment (ignatenkobrain) [inline librepo/metadata_downloader.c]: no...

File: librepo/yum.c [hunk 2]
```diff
@@ -20,32 +20,27 @@
 
 #define _POSIX_SOURCE
 #define _DEFAULT_SOURCE
+#define  BITS_IN_BYTE 8
```
Comment (ignatenkobrain) [inline librepo/yum.c]: everyone knows this, why does it need to be defined!?

**Comment by ignatenkobrain** [comment]
also split cleanup commits

File: librepo/metadata_downloader.c (hunk 1 above)
Comment (jrohel) [reply librepo/metadata_downloader.c]: No. Problem with older glibc.

**Comment by mhatina** [comment]
@ignatenkobrain what do you mean by splitting cleanup commits?

File: librepo/metadata_downloader.c [hunk 3]
```diff
@@ -87,15 +89,42 @@ lr_metadatatarget_free(LrMetadataTarget *target)
...
+
+    target->err = g_list_append(target->err, (gpointer) error_message);
+}
+
 static gboolean
 lr_setup_sigaction(struct sigaction *old_sigact,
                    gboolean interruptible,
                    GError **err)
 {
-    // Setup sighandler
```
Comment (Conan-Kudo) [inline librepo/metadata_downloader.c]: Why is this comment deleted?

File: librepo/python/metadatatarget-py.c [hunk 4]
```diff
@@ -59,8 +59,6 @@ MetadataTarget_SetThreadState(PyObject *o, PyThreadState **state)
     if (!self) return;
     self->state = state;
 
-    // XXX: Little tricky (but not so much)
-    // Set the state to the depending handle as well
```
Comment (Conan-Kudo) [inline librepo/python/metadatatarget-py.c]: Again, why delete comments?

File: librepo/python/metadatatarget-py.c [hunk 5]
```diff
@@ -81,7 +79,7 @@ check_MetadataTargetStatus(const _MetadataTargetObject *self)
 static int
 metadatatarget_progress_callback(void *data, double total_to_download, double now_downloaded)
 {
-    int ret = LR_CB_OK; // Assume everything will be ok
+    int ret = LR_CB_OK;
```
Comment (Conan-Kudo) [inline librepo/python/metadatatarget-py.c]: Why are comments being deleted?

File: librepo/python/metadatatarget-py.c [hunk 6]
```diff
@@ -101,12 +99,9 @@ metadatatarget_progress_callback(void *data, double total_to_download, double no
                                    "(Odd)", user_data, total_to_download, now_downloaded);
 
     if (!result) {
-        // Exception raised in callback leads to the abortion
-        // of whole downloading (it is considered fatal)
```
Comment (Conan-Kudo) [inline librepo/python/metadatatarget-py.c]: Deleting comments, why?

File: librepo/python/metadatatarget-py.c [hunk 7]
```diff
@@ -101,12 +99,9 @@ metadatatarget_progress_callback(void *data, double total_to_download, double no
...
         ret = LR_CB_ERROR;
     } else {
         if (result == Py_None) {
-            // Assume that None means that everything is ok
```
Comment (Conan-Kudo) [inline librepo/python/metadatatarget-py.c]: Isn't this still true?

File: librepo/yum.c (hunk 2 above)
Comment (mhatina) [reply librepo/yum.c]: Avoiding magic number.

File: librepo/python/metadatatarget-py.c (hunk 7 above)
Comment (mhatina) [reply librepo/python/metadatatarget-py.c]: Unnecessary comment.

File: librepo/python/metadatatarget-py.c (hunk 5 above)
Comment (mhatina) [reply librepo/python/metadatatarget-py.c]: You know what the line does. It doesn't need a comment.

File: librepo/python/metadatatarget-py.c (hunk 4 above)
Comment (mhatina) [reply librepo/python/metadatatarget-py.c]: Unnecessary comment.

File: librepo/metadata_downloader.c (hunk 3 above)
Comment (mhatina) [reply librepo/metadata_downloader.c]: Look at the name of function.

File: librepo/python/metadatatarget-py.c (hunk 6 above)
Comment (mhatina) [reply librepo/python/metadatatarget-py.c]: Unnecessary comment

**Comment by mhatina** [comment]
@rh-atomic-bot r+

**Pull request merged**